vaapi/vaapiimagepool.cpp

```diff
@@ -69,6 +69,7 @@ struct VaapiImagePool::ImageRecycler
         if (!image)
             return;
         m_pool->recycleID(image->getID());
+        m_pool.reset();
```

Comment:
We want to reset the pool for each ImageRecycler call so that when it destroys each Surface, it also releases the corresponding m_pool. Without the reset call, the use_count remains the same each time ImageRecycler is called.